src/DataStreams/TTLBlockInputStream.cpp

```diff
@@ -365,34 +379,24 @@ void TTLBlockInputStream::removeValuesWithExpiredColumnTTL(Block & block)
         }
         column_with_type.column = std::move(result_column);
     }
-
-    for (const String & column : columns_to_remove)
-        block.erase(column);
 }
 
 void TTLBlockInputStream::updateTTLWithDescriptions(Block & block, const TTLDescriptions & descriptions, TTLInfoMap & ttl_info_map)
 {
-    std::vector<String> columns_to_remove;
     for (const auto & ttl_entry : descriptions)
     {
         auto & new_ttl_info = ttl_info_map[ttl_entry.result_column];
         if (!block.has(ttl_entry.result_column))
-        {
-            columns_to_remove.push_back(ttl_entry.result_column);
             ttl_entry.expression->execute(block);
-        }
 
-        const IColumn * ttl_column = block.getByName(ttl_entry.result_column).column.get();
+        auto ttl_column = extractRequieredColumn(*ttl_entry.expression, block, ttl_entry.result_column);
```

Comment:
Why we use `extractRequieredColumn` if just executed the expression on the previous line?